pandas/core/generic.py

```diff
@@ -10834,7 +10834,12 @@ def _make_min_count_stat_function(cls, name, name1, name2, axis_descr, desc,
     def stat_func(self, axis=None, skipna=None, level=None, numeric_only=None,
                   min_count=0,
                   **kwargs):
-        nv.validate_stat_func(tuple(), kwargs, fname=name)
+        if name == 'sum':
+            nv.validate_sum(tuple(), kwargs)
+        elif name == 'prod':
+            nv.validate_prod(tuple(), kwargs)
```

Comment:
can we make this more generic? IOW have ``nv.validate_statu_func`` dispatch based on fname?

Reply:
I think that would make sense, but it isn't really written that way right now. Right now we just have instances of CompatValidator sitting in `function.py`, and we choose the right one to call. 

It'd be nice to have a decorator that did it for us

```python
@validate_numpy
def mean(self, ...):
    ...
```

then we don't have the duplication of the function name. But I think that's a decent sized refactor of how things are done now.

Reply:
> But I think that's a decent sized refactor of how things are done now.

Agreed.  At the time, I wrote it that way to keep things explicit, and it made it slightly easier to handle the details of each analogous `numpy` function.

That being said, I think refactoring to dispatching seems reasonable as well but would be best served for investigation and execution in a follow-up.

Reply:
ok for now. let's followup and see if can reduce need to specify names like this.